ML/AliMLModelHandler.cxx

```diff
@@ -153,6 +154,33 @@ std::string AliMLModelHandler::ImportFile(std::string path) {
     }
     return path;
   }
+
+  // check if file is on cvmfs
+  if (path.find("/cvmfs") != std::string::npos) {
+    bool checkFile = gSystem->AccessPathName(path.c_str());
+    if (checkFile) {
+      AliFatalClass(Form("Error file %s not found on CVMFS! Exit", path.data()));
+    }
+    // only skip copying for certain file types
+    std::string fileform = path.substr(path.find_last_of(".")+1,path.size());
+    if(fileform == "so" || fileform == "yml" || fileform ==  "yaml"){
+      return path;
+    }
+  }
+
+  // check if file is on cvmfs, but still requires to set the full path (so starts with PWG or AODB)
+  if (path.rfind("PWG", 0) != std::string::npos || path.rfind("AODB", 0) != std::string::npos) {
+    std::string path_cvmfs = AliDataFile::GetFileName(path);
+    if (path_cvmfs == "") {
+      AliFatalClass(Form("Error file %s not found on CVMFS! Exit", path.data()));
+    }
+    // only skip copying for certain file types
+    std::string fileform = path.substr(path.find_last_of(".")+1,path.size());
+    if(fileform == "so" || fileform == "yml" || fileform ==  "yaml"){
+      return path_cvmfs;
+    }
+    path = path_cvmfs;
+  }
```

Comment:
The previous use case can be added here, checking if `path_cvmfs` contains `root://` and defaulting to always copy the files in that case. I would add also a disclaimer like in https://github.com/alisw/AliPhysics/blob/1e729f477822aaf60ba6f6c7f74617566df97f55/PWGHF/vertexingHF/AliAnalysisTaskSEImproveITSCVMFS.cxx#L1519-L1522
Finally, looking for `"PWG"` or `"AODB"` in the path could lead to conflicts for certain paths when you try to use the code reading models stored on your machine. So some sort of protection could be added, like checking for `cvmfs:` at the beginning of the path (and then removing it) like it is done for files on alien. Maybe there are smarter solutions BTW

Reply:
Hi @lvermunt, you can have this structure by accident if you are working locally and reading the models from your machine. In any case, it is an unlikely situation.